conans/client/cmake.py

```diff
@@ -135,6 +135,12 @@ def _cmake_compiler_options(self, the_os, arch):
     def _cmake_cross_build_defines(self, the_os, os_ver):
         ret = OrderedDict()
         os_ver = get_env("CONAN_CMAKE_SYSTEM_VERSION", os_ver)
+        toolchain_file = get_env("CONAN_CMAKE_TOOLCHAIN_FILE", "")
+
+        if toolchain_file != "":
+            logger.info("Setting Cross build toolchain file: %s" % toolchain_file)
+            ret["CMAKE_TOOLCHAIN_FILE"] = toolchain_file
+            return ret
```

Comment:
I think this approach is not bad: if toolchain is defined, then, discard all remaining cross-build variables, i.e., all configuration must be defined in the toolchain file.

What do you think @lasote?